### PR TITLE
feat: canonicalize all unquoted identifier to lowercase

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

See the change in `upper_case_table_name` for examples. From now on only quoted (with any quote style) identifiers will present its case.

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by WenyXu — COMMENTED** [review]
lGtM

**Review by WenyXu — APPROVED** [review]

File: src/sql/src/parsers/create_parser.rs [hunk 1]
```diff
@@ -399,7 +405,7 @@ impl<'a> ParserContext<'a> {
     pub fn parse_column_def(&mut self) -> std::result::Result<ColumnDef, ParserError> {
         let parser = &mut self.parser;
 
-        let name = parser.parse_identifier()?;
+        let name = Self::canonicalize_identifier(parser.parse_identifier()?);
```
Comment (fengjiachun) [inline src/sql/src/parsers/create_parser.rs:408]: The method has already called canonicalize_identifier upon returning. Do we still need to call canonicalize_identifier here?

**Review by fengjiachun — APPROVED** [review]

**Review by killme2008 — APPROVED** [review]
LGTM. Note: do we need to change PromQL parser too?

**Comment by waynexia** [comment]
>do we need to change PromQL parser too?

PromQL is case-sensitive. We don't have to do this

**Pull request merged**